packages/flutter_tools/lib/src/cache.dart

```diff
@@ -143,6 +145,38 @@ class Cache {
     }
   }
 
+  /// Create a [Cache] for testing.
+  ///
+  /// Defaults to a memory file system, fake platform,
+  /// buffer logger, and no accessible artifacts.
+  /// By default, the root cache directory path is "cache".
+  @visibleForTesting
+  factory Cache.test({
+    Directory rootOverride,
+    List<ArtifactSet> artifacts,
+    Logger logger,
+    FileSystem fileSystem,
+    Platform platform,
+    ProcessManager processManager,
+  }) {
+    fileSystem ??= rootOverride?.fileSystem ?? MemoryFileSystem.test();
```

Comment:
If the `rootOverride` is passed in but not `fileSystem`, use the override file system, otherwise drop back to `MemoryFileSystem`